clusters/resource_cluster.go

```diff
@@ -323,36 +399,36 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 		// workers is changed because a resizing cluster can still serve queries
 		if isNumWorkersResizeForNonAutoscalingCluster ||
 			isAutoScalingToNonAutoscalingResize {
-			clusterInfo, err = clusters.Resize(ResizeRequest{
-				ClusterID:  clusterID,
+			_, err = clusters.Resize(ctx, compute.ResizeCluster{
+				ClusterId:  clusterId,
 				NumWorkers: cluster.NumWorkers,
 			})
+			if err != nil {
+				return err
+			}
 		} else if isAutoscaleConfigResizeForAutoscalingCluster ||
 			isNonAutoScalingToAutoscalingResize {
-			clusterInfo, err = clusters.Resize(ResizeRequest{
-				ClusterID: clusterID,
-				AutoScale: cluster.Autoscale,
+			_, err = clusters.Resize(ctx, compute.ResizeCluster{
+				ClusterId: clusterId,
+				Autoscale: cluster.Autoscale,
 			})
 		} else {
-			clusterInfo, err = clusters.Edit(cluster)
+			var editCluster compute.EditCluster
+			editCluster.ClusterId = clusterId
+			common.DataToStructPointer(d, clusterSchema, &editCluster)
+			_, err = clusters.Edit(ctx, editCluster)
 		}
 		if err != nil {
 			return err
 		}
-
-	} else {
-		clusterInfo, err = clusters.Get(clusterID)
-		if err != nil {
-			return err
-		}
```

Comment:
Did you mean to remove this branch?

Reply:
Yes, I moved it down: `clusterInfo, err = clusters.GetByClusterId(ctx, clusterId)` because we don't use clusterInfo before it. But we would want to check if we are able to successfully get the cluster before pinning / unpinning the cluster. Move it back.